src/main/scala/no/ndla/draftapi/service/StateTransitionRules.scala

```diff
@@ -107,6 +109,12 @@ trait StateTransitionRules {
       getTransition(current.status.current, to, user) match {
         case Some(t) =>
           val currentToOther = if (t.addCurrentStateToOthersOnTransition) Set(current.status.current) else Set()
+          val containsIllegalStatuses = current.status.other.intersect(t.illegalStatuses)
+          if (containsIllegalStatuses.nonEmpty) {
+            val illegalStateTransition = IllegalStatusStateTransition(
+              s"Cannot go to $to when article contains $containsIllegalStatuses")
+            return (Failure(illegalStateTransition), _ => Failure(illegalStateTransition))
```

Comment:
Er det noen grunn til at vi bruker `return` her? :rabbit:

Reply:
Ja, hvis ikke vil den eksekvere resten av koden som kommer etter if-en. Måtte evt wrappa det i en if-else for at Failure-en skal være det siste i koden og dermed bli returnert.